src/cluster_legacy.c

```diff
@@ -5419,6 +5427,28 @@ static void clusterNodeCronFreeLinkOnBufferLimitReached(clusterNode *node) {
     freeClusterLinkOnBufferLimitReached(node->inbound_link);
 }
 
+/**
+ * Compute the maximum number of connection attempts the cluster-cron
+ * loop should schedule in a single cron.
+ *
+ * We want to guarantee that every node is contacted twice within cluster node timeout.
+ */
+static long long maxConnectionAttemptsPerCron(const int nodes, const long long timeout_ms) {
+    if (nodes <= 0 || timeout_ms <= 0)
+        return 0;
+    /*
+     * We run the cron loop every 100 ms.  To reach 100 % of the nodes
+     * within the timeout, we need: ceil(nodes * 100 / timeout_ms)
+     * We use (a + b − 1) / b to give us the integer-ceil without using floating-point.
+     */
+    const long long min_nodes_for_coverage = (nodes * 100 + timeout_ms - 1) / timeout_ms;
```

Comment:
Let's add a macro here. I'm fine with ceil as well, do you see any risk here?

`#define CLUSTER_CRON_PERIOD_MS 100`